### DISCUSSION TITLE
Translated options for select field on DataObject class

### DISCUSSION THREAD

**Discussion by kursph** [pimcore/pimcore#16001] [Q&A / Support]
Hi.

I have a class with the field `salutation` as select type. I wrote a options provider which goes through the shared translations looking for a specific id:
```
private const SALUTATION_IDS = [
        'MALE' => 'salutationMale',
        'FEMALE' => 'salutationFemale',
    ];

    public function __construct(private readonly TranslatorInterface $translator)
    {
    }

    public function getOptions($context, $fieldDefinition): array
    {
        $options = [];

        foreach (self::SALUTATION_IDS as $value) {
            $options[] = [
                'key' => $this->translator->trans($value),
                'value' => $this->translator->trans($value),
            ];
        }

        return $options;
    }
```
I set the `salutation` field within a `localizedfields`:
![Bildschirm­foto 2023-09-25 um 11 52 55](https://github.com/pimcore/pimcore/assets/30214213/87cadfa6-6acc-4278-8865-9a8e8b2545de)
How can I tell the provider to translate into the corresponding language for the selected tab in the admin?
![Bildschirm­foto 2023-09-25 um 11 54 03](https://github.com/pimcore/pimcore/assets/30214213/364694db-840c-4a73-b77c-6474049a0c72)
![Bildschirm­foto 2023-09-25 um 11 54 40](https://github.com/pimcore/pimcore/assets/30214213/d942dba7-4e96-44de-8300-d58a12d20f09)
(In german it should be 'Herr' and 'Frau')

**Comment by Maksonis** [comment]
@kursph you don't need to translate key and value (value should be unique) here. Key by default can be translated in Admin Translations.

**Reply by kursph** [reply]
can you give me an example @Maksonis? I don't quite understand your approach. That would mean I just continue using `'value' => $this->translator->trans($value)`? But that already is in place and the options in the dropdown still only show the english version even for the german tab in the `localizedfields`.

**Reply by Maksonis** [reply]
@kursph sorry, I understood wrong. No idea how to solve your requirement. I guess you will need to override default logic.

**Comment by kursph** [comment]
I have to resort to using just one language in the dropdown and translate the value when being sent to the FE
